testing/trino-tests/src/test/java/io/trino/execution/EventsCollector.java

```diff
@@ -80,41 +91,59 @@ public synchronized void addQueryCompleted(QueryCompletedEvent event)
 
     public synchronized void addSplitCompleted(SplitCompletedEvent event)
     {
-        if (!eventFilters.splitCompletedFilter.test(event)) {
+        if (!eventFilters.getSplitCompletedFilter().test(event)) {
             return;
         }
         splitCompletedEvents.add(event);
         eventsLatch.countDown();
     }
 
-    public List<QueryCreatedEvent> getQueryCreatedEvents()
+    public synchronized List<QueryCreatedEvent> getQueryCreatedEvents()
```

Comment:
This is the actual fix. The writer thread may be different from the reader thread, and without the synchronization, you can get an old view of the list.